pkg/plugins/tsdb.go

```diff
@@ -189,6 +190,44 @@ type DataResponse struct {
 	Message string                     `json:"message,omitempty"`
 }
 
+// ToBackendDataResponse converts the legacy format to the standard SDK format
```

Comment:
@kylebrandt I moved this code from transform to here

Reply:
:+1: This is the pre-SSE frame reformating, so this specific part LGTM.

Reply:
Is there any benefit to return a pointer type, as this will carry with it the risk of nil-pointer exceptions? I don't think there's any perf argument either, since with so little to copy, the stack should be faster than heap.